### PR TITLE
Bnb 19 backtesting

### PR DESCRIPTION
**Pull request by YohanGH**

### CODE REVIEW

File: src/Components/StrategyControls/useStrategy.ts [hunk 1]
```diff
@@ -93,11 +96,52 @@ REASON: \${Explain why you made this decision}
 \`\`\`
   `;
 
-  const { takeTradingViewScreenshot, setScreenshot } = useScreenshot();
-
-  const runStrategyProd = async () => {
+  // Capture a screenshot
+  const handleBlobs = async (): Promise<Blob[]> => {
```
Comment (bertrandbuild) [inline src/Components/StrategyControls/useStrategy.ts:100]: this function is misleading. 
- why an array of Blob ? 
- why "handle" -> getBlob
- it's just a wrapper for "takeTradingViewScreenshot", why not use directly "takeTradingViewScreenshot" ?

File: src/Components/backtestingControls/AnalysisLauncher.tsx [hunk 2]
```diff
@@ -57,22 +57,9 @@ const AnalysisLauncher: React.FC<AnalysisLauncherProps> = ({
         disabled={isAnalyzing}
         className="bg-blue-500 text-white p-2 rounded"
       >
-        {isAnalyzing ? "Analysis in progress..." : "Start analysis"}
+        {isAnalyzing || isLoading ? "Analysis in progress..." : "Start analysis"}
```
Comment (bertrandbuild) [inline src/Components/backtestingControls/AnalysisLauncher.tsx:60]: seems like a mismatch with the var names / usage

File: src/utils/pinataUpload.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
+import { PinataSDK } from "pinata";
+import toast from "react-hot-toast";
+
+// Interface for Pinata upload response
+interface PinResponse {
```
Comment (bertrandbuild) [inline src/utils/pinataUpload.ts:5]: very good to type but I'm surprised that this type is not directly present into the sdk / type-sdk

File: src/utils/ipfs.ts [hunk 4]
```diff
@@ -1,6 +1,7 @@
 class IpfsUploadError extends Error {
```
Comment (bertrandbuild) [inline src/utils/ipfs.ts:1]: with the addition of the new pinata file, is this one still used ? 
maybe it should be deleted or we need other code updates from ipfs to pinata file

File: src/Components/backtestingControls/AnalysisLauncher.tsx (hunk 2 above)
Comment (YohanGH) [reply src/Components/backtestingControls/AnalysisLauncher.tsx:53]: modification => {isAnalyzing ? "Analysis in progress..." : "Start analysis"}

File: src/utils/ipfs.ts (hunk 4 above)
Comment (YohanGH) [reply src/utils/ipfs.ts:1]: I kept the old config for screens and the new one for url uploads.

**Pull request merged**